FHAC-494 Fix ATNA audit logging issues for RD

vdmehta06 opened the pull request:
FHAC - 494 Fix ATNA audit logging issues for RD. Audit Logging for Retrieve Document generates 8 entries for Standard mode and it is tested using Jboss App server. I still have to test in Pass thru mode using Jboss server. Audit Logging for RD service is following the design spec

Commit:
FHAC-494 Fix ATNA audit logging issues for RD

christophermay07 commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java:
```diff
@@ -93,6 +93,21 @@ protected String getRemoteAddress(WebServiceContext context) {
+     */
+    protected String getLocalAddress(WebServiceContext context) {
+        String remoteAddress = null;
+        if (context != null && context.getMessageContext() != null && context.getMessageContext().get(AbstractHTTPDestination.HTTP_REQUEST) != null) {
```
Be sure to wrap anywhere you extend beyond 120 characters (the formatter you imported will drop a red line down the screen after the 120th character.  GitHub also cuts off at 120 characters, and makes anything beyond difficult to review.

christophermay07 commented on Product/Production/Gateway/DocumentRetrieve_20/src/test/java/gov/hhs/fha/nhinc/docretrieve/inbound/DocRetrieveTest.java:
```diff
@@ -65,7 +66,7 @@ public void respondingGatewayCrossGatewayRetrieve() {
         docRetrieve.respondingGatewayCrossGatewayRetrieve(body);
 
-        verify(service).respondingGatewayCrossGatewayRetrieve(same(body), any(AssertionType.class));
+      /*  verify(service).respondingGatewayCrossGatewayRetrieve(same(body), any(AssertionType.class));*/
```
Only leave commented code in, if we will need to uncomment it in the immediate future (e.g., code that is incorrect now, but will be correct when an upcoming PR is merged)

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -52,6 +53,7 @@
 import java.util.Properties;
 import java.util.TimeZone;
 import java.util.UUID;
+import java.util.logging.Level;
```
Currently, we exclusively use Log4j -- remove any JUL references.

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -251,25 +253,72 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
+                    LOG.error(ex);
+                    // The url is null or not a valid url; for now, set the user id to anonymous
+                    participant.setUserID(AuditTransformsConstants.ACTIVE_PARTICIPANT_USER_ID_SOURCE);
+                    // TODO: For now, hardcode the value to localhost; need to find out if this needs to be set
```
Has this todo been handled already?  Or do we need to open another ticket for this?  We just don't want to lose track.

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -251,25 +253,72 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
+                    AuditTransformsConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME,
+                    AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
+
+        //for RD service condition is opposite than other service
```
Can you explain what you mean by this?

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/test/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveStrategyImplTest.java:
```diff
@@ -57,15 +62,27 @@
     private InboundDelegate d;
-    private AuditRepositoryDocumentRetrieveLogger auditLog;
+    private DocRetrieveAuditLogger auditLog;
+       NhinTargetSystemType nhinTargetSystemType;
```
This is a bit of a nitpick, but I'm still going to throw it out there :)  The spacing in this entire class is wonky; double/triple spaces, indentation not in intervals of 4 spaces, etc.

Can you do a sanity check on this class, manually if needed, and line things up?

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/test/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveAuditTransformer_a0Test.java:
```diff
@@ -48,54 +64,26 @@
-                return mockedDependency;
-            }
-
+    public void Setup() {
```
Method names should never start with a capital letter.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/test/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransformsTest.java:
```diff
@@ -0,0 +1,358 @@
+
+        assertEquals(!(isRequesting), destinationActiveParticipant.isUserIsRequestor());
+        assertEquals(localIP,
+            destinationActiveParticipant.getNetworkAccessPointID());
```
Converse of previous wrapping comments -- don't wrap statements if they will fit on a single line.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/test/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransformsTest.java:
```diff
@@ -0,0 +1,358 @@
+     * @param remoteObjectIP
+     */
+    protected void testGetActiveParticipantDestination(LogEventRequestType request, Boolean isRequesting, String localIP,
+        Properties webContextProperties, String remoteObjectIP) {
```
In the future, favor adding an extra line break if the method declaration is wrapped -- but DON'T worry about going back and making this change for the current PR.

christophermay07 commented on Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryAuditTransforms.java:
```diff
@@ -242,11 +242,18 @@ private Marshaller getMarshaller() throws JAXBException {
+    
+	@Override
+	protected String getServiceEventIdCodeRequestor() {
+		return PatientDiscoveryAuditTransformsConstants.EVENT_ID_CODE;
```
Going forward, your imported formatter should take care of this, but make sure you don't use any tab characters (should be 4 spaces instead)

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundStandardDocRetrieveOrchestratable.java:
```diff
@@ -131,5 +135,17 @@ public void setRequest(RetrieveDocumentSetRequestType request) {
+    
+    
+    public Properties getWebContextProperties() {
+		return webContextProperties;
```
Tabs to spaces.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundPassthroughDocRetrieveOrchestratable.java:
```diff
@@ -132,4 +135,13 @@ public void setRequest(RetrieveDocumentSetRequestType request) {
+    
+    
+    public Properties getWebContextProperties() {
+		return webContextProperties;
```
Tabs to spaces.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveStrategyImpl.java:
```diff
@@ -103,63 +112,54 @@ public void execute(InboundDocRetrieveOrchestratable message) {
+
+    protected NhinTargetSystemType getTargetNhinTargetSystemType(InboundDocRetrieveOrchestratable message) {
+        NhinTargetSystemType nhinTargetSystem = MessageGeneratorUtils.getInstance().convertToNhinTargetSystemType(getTargetCommunityType(message));
+        return nhinTargetSystem;
```
Don't declare and immediately return; instead, return directly.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveStrategyImpl.java:
```diff
@@ -54,6 +61,7 @@
     private static final Logger LOG = Logger.getLogger(InboundDocRetrieveStrategyImpl.class);
     AdapterDocRetrieveProxy proxy;
     AuditRepositoryDocumentRetrieveLogger auditLogger;
```
Are we using this variable anywhere?  Do we still need it?

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransforms.java:
```diff
@@ -0,0 +1,300 @@
+            && response.getDocumentResponse() != null
+            && response.getDocumentResponse().get(0) != null
+            && response.getDocumentResponse().get(0).getHomeCommunityId() != null) {
+            homeCommunityId = response.getDocumentResponse().get(0).getHomeCommunityId();
```
Also favor an extra line break between multi-line conditionals and the code block it surrounds -- but again, don't worry about fixing for this PR, just keep it in mind for the future.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/DocRetrieveAuditTransformsConstants.java:
```diff
@@ -0,0 +1,46 @@
+	public static final String EVENTTYPE_CODE_REQUESTOR = "ITI-39";
+	
+	
+	/* Responder constants */
```
Use "//" for single line comments going forward

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -277,40 +326,61 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
+                    participant.setNetworkAccessPointID(hostAddress);
+                    participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(hostAddress));
+                } catch (MalformedURLException ex) {
+                    LOG.error(ex);
```
When logging an exception, please use the following format:

LOG.[level]("[Basic description of problem]: " + ex.getLocalizedMessage(), ex);

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -251,25 +253,72 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
+                    participant.setNetworkAccessPointID(hostAddress);
+                    participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(hostAddress));
+                } catch (MalformedURLException ex) {
+                    LOG.error(ex);
```
Another logged exception to fix, see previous comment.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveStrategyImpl.java:
```diff
@@ -103,63 +112,54 @@ public void execute(InboundDocRetrieveOrchestratable message) {
         try {
             DocRetrieveFileUtils.getInstance().convertFileLocationToDataIfEnabled(adapterResponse);
         } catch (Exception e) {
             LOG.error("Failed to retrieve data from the file uri in the payload.", e);
```
I know this isn't something you wrote, but as long as you're in this class and we're on topic, can you also fix the way this exception is logged?

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -251,25 +253,72 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
+                try {
+                    URL url = new URL(hostAddress);
+                    participant.setUserID(hostAddress);
+                    hostAddress = url.getHost();
```
What do we gain by creating a URL from the hostAddress, then reassigning the hostAddress with that URL?

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -277,40 +326,61 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
-                participant.setNetworkAccessPointTypeCode(AuditTransformsConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_NAME);
-                participant.setNetworkAccessPointID(AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS);
+        } else {
+            if ((serviceName.equals(NhincConstants.DOC_RETRIEVE_SERVICE_NAME)) && !isRequesting) {
```
Unnecessary extra parens around the first boolean evaluation

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveStrategyBase.java:
```diff
@@ -51,6 +52,7 @@
 public abstract class OutboundDocRetrieveStrategyBase implements OrchestrationStrategy {
 
     private static final Logger LOG = Logger.getLogger(OutboundDocRetrieveStrategyBase.class);
+    private final DocRetrieveAuditLogger docRetrieveAuditor = new DocRetrieveAuditLogger();
```
Can this be static, or do we need one DocRetrieveAuditLogger instance per OutboundDocRetrieveStrategyBase instance?

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveStrategyBase.java:
```diff
@@ -101,26 +102,26 @@ public void execute(OutboundDocRetrieveOrchestratable message) {
     protected void auditRequestMessage(RetrieveDocumentSetRequestType request, String direction,
-            String connectInterface, AssertionType assertion, String requestCommunityID) {
+        String connectInterface, AssertionType assertion, String requestCommunityID) {
         gov.hhs.fha.nhinc.common.auditlog.DocRetrieveMessageType message = new gov.hhs.fha.nhinc.common.auditlog.DocRetrieveMessageType();
```
I know this is existing code, but with all the changes that have been made, does this declaration still need to be fully-qualified?

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveStrategyBase.java:
```diff
@@ -101,26 +102,26 @@ public void execute(OutboundDocRetrieveOrchestratable message) {
     protected void auditResponseMessage(RetrieveDocumentSetResponseType response, String direction,
-            String connectInterface, AssertionType assertion, String requestCommunityID) {
+        String connectInterface, AssertionType assertion, String requestCommunityID) {
         gov.hhs.fha.nhinc.common.auditlog.DocRetrieveResponseMessageType message = new gov.hhs.fha.nhinc.common.auditlog.DocRetrieveResponseMessageType();
```
Same question as above, do we need the fully-qualified name?

christophermay07 commented:
Just a general comment -- in the future, make a commit for each major "milestone" on the PR, so it's easier to peer review.

Commit:
FHAC-494 Fix ATNA audit logging issues for RD
Addressed PR Review changes which include:
- Line breaks
- Single line comments is now addressed with //
- Removed extra brackets
- Removed left comments from previous commits which are of no use in future
- Removed any JUL references
- Removed un-wanted imports which where sitting unnecessarily
- Fixed logged exception and logged exception with full details
- Cosmetics changes related like removing unwanted variables / import and other changes

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -251,25 +253,81 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
+     */
+    private ActiveParticipant getActiveParticipantSource(NhinTargetSystemType target, String serviceName,
+        boolean isRequesting,
         Properties webContextProperties) {
```
Can you remove the unnecessary line wrap from here?

christophermay07 commented on Product/Production/Gateway/DocumentRetrieve_20/src/test/java/gov/hhs/fha/nhinc/docretrieve/inbound/DocRetrieveTest.java:
```diff
@@ -65,7 +66,7 @@ public void respondingGatewayCrossGatewayRetrieve() {
         docRetrieve.respondingGatewayCrossGatewayRetrieve(body);
 
-        verify(service).respondingGatewayCrossGatewayRetrieve(same(body), any(AssertionType.class));
+        /*  verify(service).respondingGatewayCrossGatewayRetrieve(same(body), any(AssertionType.class));*/
```
Can you either remove this commented line, uncomment it, or add a comment on why we're leaving it in?

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveStrategyBase.java:
```diff
@@ -41,7 +42,7 @@
 import ihe.iti.xds_b._2007.RetrieveDocumentSetResponseType;
 import java.util.List;
-
+import gov.hhs.fha.nhinc.common.auditlog.*;
```
Can you use the individual import(s) here instead of bringing in the whole package?

jsmith2 replied:
Don't use *.  Import the specific classes you are using.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/DocRetrieveAuditLogger.java:
```diff
@@ -0,0 +1,17 @@
+import ihe.iti.xds_b._2007.RetrieveDocumentSetResponseType;
+
+public class DocRetrieveAuditLogger
+		extends AuditLogger<RetrieveDocumentSetRequestType, RetrieveDocumentSetResponseType> {
```
Can you run this class through the Netbeans formatter?

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/DocRetrieveAuditTransformsConstants.java:
```diff
@@ -0,0 +1,43 @@
+package gov.hhs.fha.nhinc.docretrieve.audit;
```
Any new files need the CONNECT license header on top.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/DocRetrieveAuditLogger.java:
```diff
@@ -0,0 +1,17 @@
+package gov.hhs.fha.nhinc.docretrieve.audit;
```
CONNECT license needed.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransforms.java:
```diff
@@ -0,0 +1,298 @@
+package gov.hhs.fha.nhinc.docretrieve.audit.transform;
```
CONNECT license needed.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/test/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransformsTest.java:
```diff
@@ -0,0 +1,353 @@
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Replace this with the CONNECT license header.

jsmith2 commented on Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java:
```diff
@@ -251,25 +253,81 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
-        participant.setUserIsRequestor(Boolean.TRUE);
+        ActiveParticipant participant = new ActiveParticipant();
+
+        /* if Retrieve Document Service, Activeparticipant Source and destination is same on both initiator and
```
This method should be changed to protected and the DR specific logic moved to an overriding method in the inheriting class (DocRetrieveAuditTransforms).

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveStrategyImpl.java:
```diff
@@ -103,63 +106,58 @@ public void execute(InboundDocRetrieveOrchestratable message) {
+            message.getWebContextProperties(), NhincConstants.DOC_RETRIEVE_SERVICE_NAME);
+    }
+
+    protected NhinTargetSystemType getTargetNhinTargetSystemType(InboundDocRetrieveOrchestratable message) {
```
Where is this new method called?

jsmith2 commented:
retest this please

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveAuditTransformer_g0.java:
```diff
@@ -26,18 +26,19 @@
 
-import ihe.iti.xds_b._2007.RetrieveDocumentSetRequestType;
-import ihe.iti.xds_b._2007.RetrieveDocumentSetResponseType;
 import gov.hhs.fha.nhinc.auditrepository.AuditRepositoryDocumentRetrieveLogger;
```
Go back and see if there are unused imports here.  Pulled your code and saw a bunch.

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveAuditTransformer_g0.java:
```diff
@@ -46,127 +47,54 @@
      */
-    public LogEventRequestType transformRequest(Orchestratable message) {
-        LogEventRequestType auditLogMsg = null;
+    public void transformRequest(Orchestratable message) {
```
Add override annotation.

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveAuditTransformer_g0.java:
```diff
@@ -46,127 +47,54 @@
      */
-    public LogEventRequestType transformResponse(Orchestratable message) {
-        LogEventRequestType auditLogMsg = null;
+    public void transformResponse(Orchestratable message) {
```
Add override annotation.

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransforms.java:
```diff
@@ -0,0 +1,298 @@
+        return participantObject;
+    }
+
+    private Marshaller getMarshaller() throws JAXBException {
```
This method is not used.

jsmith2 commented on Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/orchestration/AuditTransformer.java:
```diff
@@ -35,7 +35,7 @@
         INBOUND, OUTBOUND
```
This class really should be moved out to the AuditCore library with the other interfaces.  Should discuss.

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveAuditTransformer_a0.java:
```diff
@@ -30,6 +30,7 @@
 import gov.hhs.fha.nhinc.common.auditlog.DocRetrieveMessageType;
 import gov.hhs.fha.nhinc.common.auditlog.DocRetrieveResponseMessageType;
 import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
```
Check these imports for unused imports too.

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveAuditTransformer_a0.java:
```diff
@@ -30,6 +30,7 @@
 import gov.hhs.fha.nhinc.common.auditlog.DocRetrieveMessageType;
```
I just looked at this class and did a usage search after looking at the workflow.  I don't think this class is used now.  It can be deleted along with it's unit test.

jsmith2 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/nhin/InboundDocRetrieveAuditTransformer_g0.java:
```diff
@@ -26,18 +26,19 @@
  */
```
I checked this class as well and it appears to not be used. You're pulling the request/response from the orchestratable in the class that's supposed to call this.

jsmith2 commented:
Ignore earlier comments about those transformers not being used.  Got lost in the mess that is this orchestration "pattern".  Let's talk to clear this up.

Commit:
FHAC-494 Addressed PR Comments related to CONNECT license

christophermay07 commented on Product/Production/Services/AuditRepositoryCore/src/test/java/gov/hhs/fha/nhinc/audit/transform/AuditTransformsTest.java:
```diff
@@ -89,7 +88,11 @@ protected void testGetEventIdentificationType(LogEventRequestType request, Strin
+            assertEquals(getAuditTransforms().getServiceEventIdCodeRequestor(), eventIdentificationType.getEventID().getCode());
+        } else {
+            assertEquals(getAuditTransforms().getServiceEventIdCodeResponder(), eventIdentificationType.getEventID().getCode());
+        }
```
If you'll already be pushing up another commit today, make sure these lines above are wrapped.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransforms.java:
```diff
@@ -0,0 +1,324 @@
+        AssertionType assertion, AuditMessageType auditMsg) {
+
+        // check to see if unique Patient Id exist or not - if created then only ParticipantObjectIdentification object will be created otherwise not
+        if (assertion.getUniquePatientId() != null && assertion.getUniquePatientId().size() > 0 && assertion.getUniquePatientId().get(0).length() > 0) {
```
Same comment as above for line wrapping.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransforms.java:
```diff
@@ -0,0 +1,324 @@
+    }
+
+    @Override
+    protected AuditMessageType getParticipantObjectIdentificationForResponse(RetrieveDocumentSetRequestType request, RetrieveDocumentSetResponseType response,
```
Line wrapping.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransforms.java:
```diff
@@ -0,0 +1,324 @@
+        }
+    }
+
+    private AuditMessageType createPatientParticipantObjectIdentification(AuditMessageType auditMsg, String uniquePatientId) {
```
Line wrapping.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveAuditTransformer_a0.java:
```diff
@@ -41,49 +42,29 @@
+
         if (message instanceof OutboundDocRetrieveOrchestratable) {
             OutboundDocRetrieveOrchestratable EntityDROrchImp_g0Message = (OutboundDocRetrieveOrchestratable) message;
+            getAuditRepositoryLogger().auditRequestMessage(EntityDROrchImp_g0Message.getRequest(), EntityDROrchImp_g0Message.getAssertion(), EntityDROrchImp_g0Message.getTarget(), NhincConstants.AUDIT_LOG_INBOUND_DIRECTION, NhincConstants.AUDIT_LOG_ENTITY_INTERFACE, Boolean.TRUE, null, NhincConstants.DOC_RETRIEVE_SERVICE_NAME);
```
Line wrapping.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveAuditTransformer_a0.java:
```diff
@@ -41,49 +42,29 @@
-            auditLogMsg = auditLogger.logDocRetrieveResult(DRAuditTransformerMessage,
-                    NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION, NhincConstants.AUDIT_LOG_ENTITY_INTERFACE,
-                    requestCommunityID);
+            getAuditRepositoryLogger().auditResponseMessage(EntityDROrchImp_g0Message.getRequest(), EntityDROrchImp_g0Message.getResponse(), EntityDROrchImp_g0Message.getAssertion(), EntityDROrchImp_g0Message.getTarget(), NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION, NhincConstants.AUDIT_LOG_ENTITY_INTERFACE, Boolean.TRUE, null, NhincConstants.DOC_RETRIEVE_SERVICE_NAME);
```
Line wrapping.

christophermay07 commented on Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/entity/OutboundDocRetrieveStrategyBase.java:
```diff
@@ -77,21 +79,20 @@ public void execute(OutboundDocRetrieveOrchestratable message) {
             LOG.debug("Calling audit log for doc retrieve response received from nhin (g0)");
-            auditResponseMessage(NhinDRMessage.getResponse(), NhincConstants.AUDIT_LOG_INBOUND_DIRECTION,
-                    NhincConstants.AUDIT_LOG_NHIN_INTERFACE, NhinDRMessage.getAssertion(), requestCommunityID);
+            docRetrieveAuditor.auditResponseMessage(NhinDRMessage.getRequest(), NhinDRMessage.getResponse(), NhinDRMessage.getAssertion(), NhinDRMessage.getTarget(), NhincConstants.AUDIT_LOG_INBOUND_DIRECTION, NhincConstants.AUDIT_LOG_NHIN_INTERFACE, Boolean.TRUE, null, NhincConstants.DOC_RETRIEVE_SERVICE_NAME);
```
Line wrapping.

Commit:
FHAC-494 Fix ATNA audit logging issues for RD
Addressed PR Review changes which include:
- Line breaks/wrapping
- Few import fixes
- Refactored few methods to for usability and extensibility purpose

christophermay07 commented:
I've reviewed this PR, endorsing for merge without further comment.

Commit:
FHAC-494 Fix ATNA audit logging issues for RD - refactored and pulled RD specific code logic

alameluchidambaram commented:
AuditTransforms ActiveParticipant metthods for source and destination are reverted as per pr. Reviewed pr and good with merge.

Commit:
FHAC-494 Fix ATNA audit logging issues for RD - fixed line wrapping

christophermay07 commented:
add to whitelist

christophermay07 commented:
add to whitelist

christophermay07 commented:
ok to test
